mender-update/daemon/state_test.cpp

```diff
@@ -3038,6 +3054,12 @@ void StateTransitionsTestSubProcess(
 			artifact_url = http::JoinUrl(server.GetBaseUrl(), path::BaseName(artifact_path));
 		}
 
+		if (test.GetParam().device_type_mismatch) {
+			ofstream f(path::Join(tmpdir, "device_type"));
+			f << "device_type=mismatch-type\n";
+			ASSERT_IN_DEATH_TEST(f.good()) << "Failed to write mismatching device type";
+		}
```

Comment:
I would add another test case where an artifact explicitly depends on something that the devices does not provide.

Reply:
We have [separate unit tests](https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/mender-update/context_test.cpp#L519) for checking the artifact's depends vs. what the device provides with pretty good coverage. This test is really about how the mismatch is handled in the state machine, it doesn't matter what doesn't match.

Reply:
Okey :+1: